lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -168,6 +244,276 @@ def partition_pdf(
         LogTime("finish", point=True)
         return deformable_layout
 
+    def _partition_pdf_batched(
+        self,
+        file: BinaryIO,
+        threshold: float = 0.4,
+        use_ocr=False,
+        ocr_images=False,
+        ocr_tables=False,
+        extract_table_structure=False,
+        table_structure_extractor=None,
+        extract_images=False,
+        batch_size: int = 1,
+    ) -> List[List["Element"]]:
+        LogTime("partition_start", point=True)
+        with tempfile.NamedTemporaryFile(prefix="detr-pdf-input-") as pdffile:
+            with LogTime("write_pdf"):
+                data = file.read()
+                data_len = len(data)
+                pdffile.write(data)
+                data = b""
+                pdffile.flush()
+                logging.info(f"Wrote {pdffile.name}")
+            stat = os.stat(pdffile.name)
+            assert stat.st_size == data_len
+            assert os.path.isfile(pdffile.name), f"missing {pdffile.name}"
+            return self._partition_pdf_batched_named(
+                pdffile.name,
+                threshold,
+                use_ocr,
+                ocr_images,
+                ocr_tables,
+                extract_table_structure,
+                table_structure_extractor,
+                extract_images,
+                batch_size,
+            )
+
+    def _partition_pdf_batched_named(
+        self,
+        filename: str,
+        threshold: float = 0.4,
+        use_ocr=False,
+        ocr_images=False,
+        ocr_tables=False,
+        extract_table_structure=False,
+        table_structure_extractor=None,
+        extract_images=False,
+        batch_size: int = 1,
+    ) -> List[List["Element"]]:
+        if not table_structure_extractor:
+            table_structure_extractor = DEFAULT_TABLE_STRUCTURE_EXTRACTOR(device=self.device)
+
+        pdfminer = None
+        exec = ProcessPoolExecutor(max_workers=1)
+        if not use_ocr:
+            with LogTime("start_pdfminer", log_start=True):
+                pdfminer = exec.submit(self._run_pdfminer, filename)
+
+        deformable_layout = []
+        if tracemalloc.is_tracing():
+            before = tracemalloc.take_snapshot()
+        for i in self._batch_pdf_convert(filename, batch_size):
+            parts = self.process_batch(
+                i,
```

Comment:
there's a bunch of flags we pass around, named arguments are probably useful so we don't mess up

Reply:
Done. Didn't seem very useful to me, but we'll see.